### ISSUE TITLE
Fix false positives for negative numbers in function-calc-no-invalid

### ISSUE THREAD

**Issue by saschask** [stylelint/stylelint#3912]
> Clearly describe the bug

There are false positives for negative numbers inside the ``calc()`` statement. Where inside the ``calc()`` a negative number is used and whether it has a unit or not does not matter.

> Which rule, if any, is the bug related to?

``function-calc-no-invalid``

> What CSS is needed to reproduce the bug?

```css
a {
  margin-left: calc(-6 * 20px);
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "function-calc-no-invalid": true
  }
}
```

> Which version of stylelint are you using?

``9.10.0``

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with ``stylelint "source/**/*.css"``

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warning is flagged:

```shell
39:42  ×  Expected a valid expression   function-calc-no-invalid
```

**Comment by jeddy3** [comment]
@saschask Thanks for the report and for using the template.

I can reproduce this false positive.
